src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+
+                        List<BoundStatement> precedingStatements = new List<BoundStatement>();
+                        for (int j = 0; j < i; j++)
+                        {
+                            precedingStatements.Add(statements[j]);
+                        }
+                        var followingStatements = ImmutableArray.Create(
+                            statements, i + 1,
+                            statements.Length - i - 1);
+
+                        var localDeclarations = ImmutableArray.Create(localDeclaration);
+
+                        BoundBlock innerBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            statements: followingStatements
+                            );
+
+                        BoundUsingStatement boundUsing = new BoundUsingStatement(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            declarationsOpt: new BoundMultipleLocalDeclarations(
+                                localDeclaration.Syntax,
+                                localDeclarations),
+                            expressionOpt: null,
+                            iDisposableConversion: Conversion.Identity,
+                            disposeMethodOpt: null,
+                            body: innerBlock
+                            );
+                        precedingStatements.Add(boundUsing);
+
+                        BoundBlock outermostBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: node.Locals,
+                            statements: precedingStatements.ToImmutableArray<BoundStatement>());
+
+                        return outermostBlock;
+
+                    }
+                }
+                else if (statements[i] is BoundMultipleLocalDeclarations boundMultiple)
+                {
+                    if (boundMultiple.LocalDeclarations.Any())
+                    {
+                        if (boundMultiple.LocalDeclarations[0].LocalSymbol.IsUsing)
+                        {
+                            List<BoundStatement> precedingStatements = new List<BoundStatement>();
+                            for (int j = 0; j < i; j++)
+                            {
+                                precedingStatements.Add(statements[j]);
+                            }
+                            List<BoundStatement> followingStatements = new List<BoundStatement>();
+                            for (int k = i + 1; i < statements.Length; k++)
+                                followingStatements.Add(statements[k]);
+                            return LowerBoundMultipleLocalDeclarationUsingVar(boundMultiple, node.Locals, precedingStatements, followingStatements.ToImmutableArray());
```

Comment:
I think an early return here is also going to drop subsequent multiple declarations.